modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java

```diff
@@ -0,0 +1,343 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.PaginationBarDelta;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.PaginationBarLabels;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.taglib.util.TagResourceBundleUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Julien Castelain
+ */
+public class PaginationBarTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		if (_activeDelta == null) {
+			PaginationBarDelta paginationBarDelta = getDeltas().get(0);
+
+			Integer label = (Integer)paginationBarDelta.get("label");
```

Comment:
Label is strange name for an Integer...

Reply:
I agree, but it's based on the [TypeScript](https://github.com/liferay/clay/blob/64c20ca9076e625bf215387e53d3b91ce744e142/packages/clay-pagination-bar/src/PaginationBarWithBasicItems.tsx#L31-L41) code, I couldn't come with anything better